src/App.tsx

```diff
@@ -1,61 +1,137 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { User } from './type/User';
+import { Todo } from './type/Todo';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [userEntered, setUser] = useState('');
+  const [title, setTitle] = useState('');
+  const [userError, setUserError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+  const [addTodo, setAddTodo] = useState(todos);
+
+  const addNewTodo = () => {
+    if (!userEntered) {
+      setUserError(true);
+    }
+
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (!userEntered || !title) {
+      return;
+    }
+
+    const newTodo = {
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
+      userId: +userEntered,
+      title,
+      completed: false,
+      user: getUser(+userEntered),
+    };
+
+    setAddTodo((state) => (
+      [
+        ...state,
+        newTodo,
+      ]));
```

Comment:
naming: it will be obvious that you're just appending a newTodo
```suggestion
    setAddTodo((prevTodos) => (
      [
        ...prevTodos,
        newTodo,
      ]));
```